File: exercism/kotlin/isogram/src/main/kotlin/Isogram.kt
object Isogram {

    // return true is input is isogram
    fun isIsogram(input: String): Boolean {
        return input.toLowerCase().toCharArray()
                .filter { "[a-z]".toRegex().matches("$it") }
                .let{ it.toSet().size == it.size}
    }

    // fun isIsogram(input: String) = input.filter(Char::isLetter).map(Char::toLowerCase).let {
    //     it.toSet().size == it.size
    // }
}

File: exercism/go/isogram/isogram.go
package isogram

import (
	"unicode"
)

// IsIsogram returns true is isogram
func IsIsogram(input string) bool {

	m := make(map[rune]bool)

	for _, ch := range input {
		if ch == ' ' || ch == '-' {
			continue
		}
		lowerCh := unicode.ToLower(ch)

		if m[lowerCh] {
			return false
		}
		m[lowerCh] = true
	}
	return true
}
